### PR TITLE
Update sun.py to use GeocentricTrueEcliptic

### PR DESCRIPTION
**Pull request by kathivashkiv**
`GeocentricMeanEcliptic` coordinates were replaced by `GeocentricTrueEcliptic`
coordinates in `apparent_longitude()` and `apparent_latitude()` functions
due to changes in Astropy4.0
(#4000 issue)

Function `apparent_longitude()` contains explicit adding of nutation:
https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/sun.py#L202, 
which to some extent repeats the change added to the transformation functions for getting to/from ecliptic systems in https://github.com/astropy/astropy/pull/10129
(https://github.com/astropy/astropy/blob/master/astropy/coordinates/builtin_frames/ecliptic_transforms.py#L44)

Therefore this correction in `apparent_longitude()` was removed (after this removal all existing in `coordinates` module tests pass).
Can you review this calculation change please?

Edit by @ayshih : not to be backported

### CODE REVIEW

**Comment by kakirastern** [comment]
@kathivashkiv You can ignore the coverage tests for now. No need to worry about these.

**Comment by kakirastern** [comment]
Great work, BTW

**Comment by kakirastern** [comment]
When the PR is ready for a review please do let us know by pinging either here or on the IRC (Element.io) channel

**Comment by kathivashkiv** [comment]
> When the PR is ready for a review please do let us know by pinging either here or on the IRC (Element.io) channel

OK) Tell me please who can I ping in such cases?

**Comment by kakirastern** [comment]
Comme moi or like @Cadair @nabobalis or @ayshih

**Comment by kakirastern** [comment]
I will assign you someone to review a PR then if no one is on call/duty

File: sunpy/coordinates/sun.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     Distance,
     GeocentricMeanEcliptic,
     HeliocentricMeanEcliptic,
+    GeocentricTrueEcliptic,
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:15]:
```suggestion
    GeocentricTrueEcliptic,
    HeliocentricMeanEcliptic,
```

File: sunpy/coordinates/sun.py [hunk 2]
```diff
@@ -196,12 +197,11 @@ def apparent_longitude(t='now'):
     """
     time = parse_time(t)
     sun = SkyCoord(0*u.deg, 0*u.deg, 0*u.AU, frame='hcrs', obstime=time)
-    coord = sun.transform_to(GeocentricMeanEcliptic(equinox=time))
+    coord = sun.transform_to(GeocentricTrueEcliptic(equinox=time))
 
     # Astropy's GeocentricMeanEcliptic already includes aberration, so only add nutation
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:202]:
```suggestion
```

File: sunpy/coordinates/sun.py [hunk 3]
```diff
@@ -196,12 +197,11 @@ def apparent_longitude(t='now'):
...
-    jd1, jd2 = get_jd12(time, 'tt')
-    nut_lon, _ = erfa.nut06a(jd1, jd2)*u.radian
-    lon = coord.lon + nut_lon
+    # change to "Astropy's GeocentricTrueEcliptic already includes both aberration and nutation"?
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:203]:
```suggestion
    # Astropy's GeocentricTrueEcliptic includes both aberration and nutation
```

File: sunpy/coordinates/sun.py [hunk 4]
```diff
@@ -241,9 +241,10 @@ def apparent_latitude(t='now'):
     """
     time = parse_time(t)
     sun = SkyCoord(0*u.deg, 0*u.deg, 0*u.AU, frame='hcrs', obstime=time)
-    coord = sun.transform_to(GeocentricMeanEcliptic(equinox=time))
+    coord = sun.transform_to(GeocentricTrueEcliptic(equinox=time))
 
     # Astropy's GeocentricMeanEcliptic does not include nutation, but the contribution is negligible
+    # change to "Astropy's GeocentricTrueEcliptic already includes both aberration and nutation"?
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:247]:
```suggestion
    # Astropy's GeocentricTrueEcliptic includes both aberration and nutation
```

**Review by ayshih — CHANGES_REQUESTED** [review]
Please add a changelog entry, categorized under `trivial`

File: changelog/4584.trivial.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+`~sunpy.coordinates.sun.apparent_longitude()` and `~sunpy.coordinates.sun.apparent_latitude()` functions will now use `~astropy.coordinates.GeocentricTrueEcliptic` coordinates
```
Comment (ayshih) [inline changelog/4584.trivial.rst:1]:
```suggestion
`sunpy.sun` functions now make use of the `~astropy.coordinates.GeocentricTrueEcliptic` frame to simplify internal calculations, but the returned values are unchanged.
```

**Review by ayshih — CHANGES_REQUESTED** [review]
I edited the changelog entry, but otherwise this should be good

**Review by dstansby — APPROVED** [review]

**Review by ayshih — APPROVED** [review]
Test failures are unrelated.

**Comment by ayshih** [comment]
Squashed via GitHub.  Thanks for the PR!

**Pull request merged**